doc-content/shared-kie-docs/src/main/asciidoc/Commands/runtime-commands-samples-ref.adoc

```diff
@@ -0,0 +1,1210 @@
+                    "age": 25
+                  }
+                }
+              ],
+              "key": "objects"
+            }
+          ],
+          "facts": []
+        }
+      }
+    }
+  ]
+}
+----
+--
+InsertElementsCommand::
++
+--
+Inserts a list of objects into the KIE session.
+
+.Command attributes
+[cols="30%,50%,20%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Requirement
+
+|`objects`
+|The list of objects to be inserted into the KIE session
+|Required
+
+|`out-identifier`
+|ID of the `FactHandle` created from the object insertion and added to the execution results
+|Optional
+
+|`return-object`
+|Boolean to determine whether the object must be returned in the execution results. Default value: `true`.
+|Optional
+
+|`entry-point`
+|Entry point for the insertion
+|Optional
+|===
+
+.Example JSON request body
+[source,json]
+----
+{
+  "commands": [ {
+    "insert-elements": {
+        "objects": [
+            {
+                "containedObject": {
+                    "@class": "org.drools.compiler.test.Person",
+                    "age": 25,
+                    "name": "john"
+                }
+            },
+            {
+                "containedObject": {
+                    "@class": "Person",
+                    "age": 35,
+                    "name": "sarah"
+                }
+            }
+        ]
+    }
+  }
+]
+}
+----
+
+.Example Java command
+[source,java]
+----
+List<Object> objects = new ArrayList<Object>();
+objects.add(new Person("john", 25));
+objects.add(new Person("sarah", 35));
+
+Command insertElementsCommand = CommandFactory.newInsertElements(objects);
+----
+
+.Example server response (JSON)
+[source,json]
+----
+{
+  "response": [
+    {
+      "type": "SUCCESS",
+      "msg": "Container command-script-container successfully called.",
+      "result": {
+        "execution-results": {
+          "results": [],
+          "facts": [
+            {
+              "value": {
+                "org.drools.core.common.DefaultFactHandle": {
+                  "external-form": "0:4:436792766:-2127720265:4:DEFAULT:NON_TRAIT:java.util.LinkedHashMap"
+                }
+              },
+              "key": "john"
+            },
+            {
+              "value": {
+                "org.drools.core.common.DefaultFactHandle": {
+                  "external-form": "0:4:436792766:-2127720266:4:DEFAULT:NON_TRAIT:java.util.LinkedHashMap"
+                }
+              },
+              "key": "sarah"
+            }
+          ]
+        }
+      }
+    }
+  ]
+}
+----
+--
+FireAllRulesCommand::
++
+--
+Executes all rules in the KIE session.
+
+.Command attributes
+[cols="30%,50%,20%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Requirement
+
+|`max`
+|Maximum number of rules to be executed. The default is `-1` and does not put any restriction on execution.
+|Optional
+
+|`out-identifier`
+|The number of rules fired on the execution results.
```

Comment:
Mmm, yes, good point. I changed to "ID to be used for the fired rules in execution results." That work?

Reply:
Better, but to be more precise it will return just a total number of rules which were fired. So something like "ID to be used for the fired rules **count** in execution results." would be better.

Reply:
@MarianMacik I'm confused. The example `out-identifier` value in the code snippets is not a number, but is `firedActivations`. Look in the examples that follow this spot. In other commands that user `out-identifier`, the `out-identifier` is similarly an ID. Search `out-identifier` in the rendered doc preview to see what I mean.

Reply:
@sterobin maybe we misunderstood each other. But `out-identifier` in this case is an ID under which you can obtain from the execution results a total number of rule activations the command has invoked. So in the example I can obtain total number of fired rules (which represents a value) under the key `firedActivations`. My whole point was that you wrote `ID to be used for the fired rules in execution results.` - but it doesn't map fired rules themselves under that ID, rather it maps a total number of executions, e.g. 5. Is it more clear now?